src/main/java/uk/gov/hmcts/reform/sscs/service/ccdupdate/CcdLocationUpdateService.java

```diff
@@ -0,0 +1,109 @@
+package uk.gov.hmcts.reform.sscs.service.ccdupdate;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.sscs.ccd.domain.*;
+import uk.gov.hmcts.reform.sscs.exception.UpdateCaseException;
+import uk.gov.hmcts.reform.sscs.model.VenueDetails;
+import uk.gov.hmcts.reform.sscs.model.messaging.HmcMessage;
+import uk.gov.hmcts.reform.sscs.service.venue.VenueRpcDetailsService;
+
+import java.util.ArrayList;
+import java.util.List;
+import javax.validation.Valid;
+
+import static java.util.Objects.isNull;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class CcdLocationUpdateService {
+
+    public static final String TEMPLATE_UPDATE_VENUE_ERROR = "Failed to update venue for Case Id %s:%n";
+
+    private final VenueRpcDetailsService venueRpcDetailsService;
+
+    public void updateVenue(HmcMessage hmcMessage, @Valid SscsCaseData sscsCaseData) throws UpdateCaseException {
+
+        Hearing oldHearing = getHearingFromCaseData(hmcMessage, sscsCaseData);
+        if (isNull(oldHearing)) {
+            UpdateCaseException exc = new UpdateCaseException(
+                String.format(TEMPLATE_UPDATE_VENUE_ERROR + "Could not find hearing with Hearing Id: %s",
+                        sscsCaseData.getCcdCaseId(), hmcMessage.getHearingID()));
+            log.error(exc.getMessage(), exc);
+            throw exc;
+        }
+
+        String updatedVenueId = hmcMessage.getHearingUpdate().getHearingVenueID();
+        Venue venue = findVenue(updatedVenueId);
+        if (isNull(venue)) {
+
+            UpdateCaseException exc = new UpdateCaseException(
+                String.format(TEMPLATE_UPDATE_VENUE_ERROR + "Could not find venue with Epims Id %s ",
+                        sscsCaseData.getCcdCaseId(), updatedVenueId));
+            log.error(exc.getMessage(), exc);
+
+            throw exc;
+        }
+
+        HearingDetails hearingDetails = oldHearing.getValue();
```

Comment:
optional: Is it possible to change this to `oldHearing.getHearingDetails()` ?

Reply:
Nope, value is needed for collections in ccd, we now use a wrapper for it but this class uses the old method